app/assets/javascripts/topology.js

```diff
@@ -0,0 +1,79 @@
+var React = require('react');
+
+var Topology = React.createClass({
+  DEFAULT_DELTA: '...',
+
+  getInitialState: function() {
+    return { delta: this.DEFAULT_DELTA };
+  },
+  processDeltas: function(deltas) {
+    console.log('parse deltas', deltas);
+    var healthyPartitions = 0,
+        unhealthyPartitions = 0,
+        totalDelta = 0;
+
+
+    deltas.forEach(function(delta) {
+      var storm = delta.storm;
+      if ( storm === null ) {
+        unhealthyPartitions++;
+        storm = 0;
+      } else {
+        unhealthyPartitions++;
+      }
+      totalDelta = delta.current - storm;
+    });
+    this.setState({ delta: totalDelta, healthy: healthyPartitions, unhealthy: unhealthyPartitions, error: false, message: '' });
+  },
+  refreshDelta: function() {
+    console.log('refreshing delta');
+    var r = new XMLHttpRequest();
+    var query = "topic=" + this.props.topic + "&toporoot=" + this.props.root;
+    r.open("GET", "/api/status?" + query, true);
+    r.onreadystatechange = function () {
+      if ( r.readyState != 4 ) { return; }
+      if ( r.status == 500 ) {
+        this.setState({ delta: this.DEFAULT_DELTA, error: true, message: 'Capillary returned a 500 error. Check ZK configuration?' });
+        return;
+      }
+      try {
+        this.processDeltas(JSON.parse(r.responseText));
+      }
+      catch(err) {
+        console.error(err);
+        this.setState({ delta: this.DEFAULT_DELTA, error: true, message: r.responseText });
+      }
+    }.bind(this);
+    r.send()
+  },
+  componentWillMount: function() {
+    //this.refreshTimer = window.setInterval(this.refreshDelta, this.props.refreshInterval);
+    this.refreshDelta();
+  },
+  componentWillUnmount: function() {
```

Comment:
Eventually we will want polling and automatic updates here, but I wanted to make sure that it works first and we can talk about intervals.